src/components/icon/icon.ts

```diff
@@ -0,0 +1,209 @@
+  @Input() fontIcon: string;
+  @Input() alt: string;
+
+  @Input('aria-label') ariaLabelFromParent: string = '';
+
+  private _previousFontSetClass: string;
+  private _previousFontIconClass: string;
+
+  constructor(
+      private _element: ElementRef,
+      private _renderer: Renderer,
+      private _changeDetectorRef: ChangeDetectorRef,
+      private _mdIconRegistry: MdIconRegistry) {
+  }
+
+  /**
+   * Splits an svgIcon binding value into its icon set and icon name components.
+   * Returns a 2-element array of [(icon set), (icon name)].
+   * The separator for the two fields is ':'. If there is no separator, an empty
+   * string is returned for the icon set and the entire value is returned for
+   * the icon name. If the argument is falsy, returns an array of two empty strings.
+   * Examples:
+   *   'social:cake' -> ['social', 'cake']
+   *   'penguin' -> ['', 'penguin']
+   *   null -> ['', '']
+   */
+  private _splitIconName(iconName: string): [string, string] {
+    if (!iconName) {
+      return ['', ''];
+    }
+    const separatorIndex = this.svgIcon.indexOf(':');
+    if (separatorIndex == -1) {
+      return ['', iconName];
+    }
+    return [iconName.substring(0, separatorIndex), iconName.substring(separatorIndex + 1)];
+  }
+
+  ngOnChanges(changes: { [propertyName: string]: SimpleChange }) {
+    const changedInputs = Object.keys(changes);
+    // Only update the inline SVG icon if the inputs changed, to avoid unnecessary DOM operations.
+    if (changedInputs.indexOf('svgIcon') != -1 || changedInputs.indexOf('svgSrc') != -1) {
+      if (this.svgIcon) {
+        const [namespace, iconName] = this._splitIconName(this.svgIcon);
+        this._mdIconRegistry.getNamedSvgIcon(iconName, namespace).subscribe(
+            svg => this._setSvgElement(svg),
+            (err: any) => console.log(`Error retrieving icon: ${err}`));
+      } else if (this.svgSrc) {
+        this._mdIconRegistry.getSvgIconFromUrl(this.svgSrc).subscribe(
+            svg => this._setSvgElement(svg),
+            (err: any) => console.log(`Error retrieving icon: ${err}`));
+      }
+    }
+    if (this._usingFontIcon()) {
+      this._updateFontIconClasses();
+    }
+    this._updateAriaLabel();
+  }
+
+  ngOnInit() {
+    // Update font classes because ngOnChanges won't be called if none of the inputs are present,
+    // e.g. <md-icon>arrow</md-icon>. In this case we need to add a CSS class for the default font.
+    if (this._usingFontIcon()) {
+      this._updateFontIconClasses();
+    }
+  }
+
+  ngAfterContentChecked() {
+    // Update aria label here because it may depend on the projected text content.
+    // (e.g. <md-icon>home</md-icon> should use 'home').
+    this._updateAriaLabel();
+  }
+
+  private _updateAriaLabel() {
+      const ariaLabel = this._getAriaLabel();
+      if (ariaLabel) {
+        this._renderer.setElementAttribute(this._element.nativeElement, 'aria-label', ariaLabel);
+        this._changeDetectorRef.detectChanges();
```

Comment:
I'm reasonably sure that calling `detectChanges` here is going to be problematic, especially since it's being done from other lifecycle events. Why is the call necessary?